src/main/java/com/couchbase/lite/store/ForestDBStore.java

```diff
@@ -1208,8 +1208,8 @@ private boolean saveForest(Document doc,
         if(!isWinner)
             // TODO
             ;
-        if (properties != null && properties.containsKey("type"))
-            doc.setType((String)properties.get("type"));
+        if (properties != null && properties.containsKey("type") && properties.get("type") instanceof String)
+            doc.setType((String) properties.get("type"));
```

Comment:
If properties.get("type") == null, do we need to set null to the type of the doc?